File: solution/kotlin/day23/src/test/kotlin/tripservicekata/user/UserBuilder.kt
package tripservicekata.user

import org.craftedsw.tripservicekata.trip.Trip
import org.craftedsw.tripservicekata.user.User


class UserBuilder {
    private var friends: List<User> = emptyList()
    private var trips: List<Trip> = emptyList()
    fun friendsWith(vararg friends: User): UserBuilder {
        this.friends = friends.toList()
        return this
    }

    fun travelledTo(vararg trips: Trip): UserBuilder {
        this.trips = trips.toList()
        return this
    }

    fun build(): User = User(trips, friends)

    companion object {
        fun aUser(): UserBuilder = UserBuilder()
    }
}

File: solution/ts/day23/tests/user/userBuilder.ts
import Trip from "../../src/trip/Trip";
import {User} from "../../src/user/User";

export class UserBuilder {
    private friends: User[] = [];
    private trips: Trip[] = [];

    static aUser(): UserBuilder {
        return new UserBuilder();
    }

    friendsWith(...friends: User[]): UserBuilder {
        this.friends = friends;
        return this;
    }

    travelledTo(...trips: Trip[]): UserBuilder {
        this.trips = trips;
        return this;
    }

    build(): User {
        return new User(this.trips, this.friends);
    }
}